main/src/ca/uwaterloo/flix/language/phase/TypeReconstruction.scala

```diff
@@ -428,7 +428,7 @@ object TypeReconstruction {
         case Type.Cst(TypeConstructor.JvmMethod(method), loc) =>
           TypedAst.Expr.InvokeMethod(method, e, es, returnTpe, eff, loc)
         case _ =>
-          throw InternalCompilerException(s"unexpected type: ${methodTpe}.", loc)
+          TypedAst.Expr.Error(TypeError.UnresolvedMethod(methodTpe.toString, loc), methodTpe, eff)
```

Comment:
This is fine as a first step, but later we'll need this error to be reflected in a Validation as well.

Reply:
Actually this error cannot really happen. It is a strange grayzone. It only happens because we may apply a partial subst to an expression.